src/hotspot/share/services/heapDumper.hpp

```diff
@@ -90,6 +79,11 @@ class HeapDumper : public StackObj {
 
   static void dump_heap_from_oome()    NOT_SERVICES_RETURN;
 
+  // Parallel thread number for heap dump, initialize based on active processor count.
+  static uint default_num_of_dump_threads() {
```

Comment:
Heapdump is executed at safepoint, why don't we use all CPU cores?

Reply:
This is derived from https://github.com/openjdk/jdk/pull/2261/files#diff-5373ebf2d7b4e9c8fcbb188dd0d7193d4af7e3e021de7e3e4dd61c9bb5803edfR249

Experiments show that using all CPUs does not achieve the best performance.